### PR TITLE
P2752R3 Static storage for braced initializers

### LINKED ISSUE #6284: [2023-06 CWG Motion 9] P2752R3 Static storage for braced initializers
https://wiki.edg.com/pub/Wg21varna/StrawPolls/P2752R3.html

### PR DESCRIPTION
**Pull request by jensmaurer**
Fixes #6284
Fixes cplusplus/papers#1442

### CODE REVIEW

**Comment by cpplearner** [comment]
Does it make sense to bump the `__cpp_initializer_lists` feature-test macro?

File: source/compatibility.tex [hunk 1]
```diff
@@ -1,6 +1,36 @@
...
+\effect
+Valid \CppXXIII{} code
+that relies on the result of pointer comparison between backing arrays
+may change behavior.
+For example:
+\begin{example}
+bool ne(std::initializer_list<int> a, std::initializer_list<int> b) {
+  return a.begin() != b.begin() + 1;
+}
+bool b = ne({2,3}, {1,2,3}); // unspecified result; previously false
```
Comment (JohelEGP) [inline source/compatibility.tex:31]:
```suggestion
bool b = ne({2,3}, {1,2,3}); // unspecified result; previously \tcode{false}
```

**Review by JohelEGP — CHANGES_REQUESTED** [review]
I'm using <https://isocpp.org/files/papers/D2752R3.html> to pre-review
(there are sufficient differences that I'll need to wait for the mailing to confirm).

**Comment by jensmaurer** [comment]
> Does it make sense to bump the `__cpp_initializer_lists` feature-test macro?

This is a pure optimization enabler; I don't think it warrants a "feature" announcement.

**Review by JohelEGP — APPROVED** [review]
> (there are sufficient differences that I'll need to wait for the mailing to confirm).

It was actually alright.
The wording in the generated PDF looks as was proposed.
But it seems there was a change in the context
and the paper's PR wasn't rebased.

**Comment by tkoeppe** [comment]
It seems that the paper is just presenting the status quo in a rather loose and lacunary fashion. The PR looks good. Thanks for checking!

**Pull request merged**